packages/flutter/lib/src/painting/image_provider.dart

```diff
@@ -510,25 +517,49 @@ class NetworkImage extends ImageProvider<NetworkImage> {
     );
   }
 
-  static final HttpClient _httpClient = HttpClient();
-
-  Future<ui.Codec> _loadAsync(NetworkImage key) async {
-    assert(key == this);
-
-    final Uri resolved = Uri.base.resolve(key.url);
-    final HttpClientRequest request = await _httpClient.getUrl(resolved);
-    headers?.forEach((String name, String value) {
-      request.headers.add(name, value);
-    });
-    final HttpClientResponse response = await request.close();
-    if (response.statusCode != HttpStatus.ok)
-      throw Exception('HTTP request failed, statusCode: ${response?.statusCode}, $resolved');
+  /// Do not access this field directly; use [_httpClient] instead.
+  static final HttpClient _sharedHttpClient = HttpClient()..autoUncompress = false;
```

Comment:
you turn off autoUncompress here, but you only support gzip on the other side. HTTP servers can use a variety of compression algorithms other than gzip.

Reply:
(You should make sure to set the `TE` header to whatever it is you support. Also, consider `Accept-Encoding` and `Content-Encoding`, which are different than `TE` and `Transfer-Encoding`.)

Reply:
HttpClient documents this property as only respecting gzip -- https://api.dartlang.org/stable/2.3.0/dart-io/HttpClient/autoUncompress.html